exe/Main.hs

```diff
@@ -62,6 +84,33 @@ ghcideVersion = do
              <> ") (PATH: " <> path <> ")"
              <> gitHashSection
 
+runWithDb :: Maybe DynFlags -> FilePath -> (HieDb -> HieWriterChan -> IO ()) -> IO ()
+runWithDb df fp k =
+  withHieDb fp $ \writedb' -> do
+    let writedb = writedb' { getDbDynFlags = df }
+    execute_ (getConn writedb) "PRAGMA journal_mode=WAL;"
+    -- setTrace (getConn writedb) (Just $ T.appendFile "/tmp/sqltrace" . (<>"\n"))
+    initConn writedb
+    chan <- newChan
+    race_ (writerThread writedb chan) $ withHieDb fp $ \readdb' -> do
+      let readdb = readdb' { getDbDynFlags = df }
+      -- setTrace (getConn readdb) (Just $ T.appendFile "/tmp/sqltrace" . (<>"\n"))
+      k readdb chan
+  where
+    writerThread db chan =forever $ do
+      k <- readChan chan
+      k db `catch` \e@SQLError{} -> do
+        hPutStrLn stderr $ "Error in worker, ignoring: " ++ show e
+        exitWith $ ExitFailure 1
+
+getHieDbLoc :: FilePath -> IO FilePath
+getHieDbLoc dir = do
+  let db = dirHash++"-"++takeBaseName dir++"-"++VERSION_ghc <.> "hiedb"
+      dirHash = B.unpack $ encode $ H.hash $ B.pack dir
+  cDir <- IO.getXdgDirectory IO.XdgCache cacheDir
+  createDirectoryIfMissing True cDir
+  pure (cDir </> db)
```

Comment:
Since `ghcide` now integrates the `hiedb` command line, I would say the way to interact with this database is via the `ghcide hiedb ...` subcommands.

So this lives in the usual ghcide cache directory, along with the `.hi`/`.hie` files.

Reply:
oh yeah, that's what I meant.